Mathlib/GroupTheory/Perm/Cycle/Factors.lean

```diff
@@ -0,0 +1,694 @@
+/-
+Copyright (c) 2019 Chris Hughes. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Chris Hughes, Yaël Dillies
+-/
+
+import Mathlib.Algebra.Module.BigOperators
+import Mathlib.Data.Finset.NoncommProd
+import Mathlib.Data.Fintype.Perm
+import Mathlib.Data.Int.ModEq
+import Mathlib.GroupTheory.Perm.List
+import Mathlib.GroupTheory.Perm.Sign
+import Mathlib.Logic.Equiv.Fintype
+
+import Mathlib.GroupTheory.Perm.Cycle.Basic
+
+#align_import group_theory.perm.cycle.basic from "leanprover-community/mathlib"@"e8638a0fcaf73e4500469f368ef9494e495099b3"
+
+/-!
+# Cycle factors of a permutation
+
+Let `β` be a `Fintype` and `f : Equiv.Perm β`.
+
+* `Equiv.Perm.cycleOf`: `f.cycleOf x` is the cycle of `f` that `x` belongs to.
+* `Equiv.Perm.cycleFactors`: `f.cycleFactors` is a list of disjoint cyclic permutations that
+  multiply to `f`.
+
```

Comment:
Sorry, this is unrelated to the split, but I've just noticed something is wrong with our generality
```suggestion

## TODO

Generalise `Equiv.Perm.cycleOf` to arbitrary types.
```